course03/miguel/project/spaceimpact.cc

```diff
@@ -0,0 +1,51 @@
+#include "spaceimpact.h"
+
+#include <cinder/app/App.h>
+#include <cinder/gl/gl.h>
+
+namespace SpaceImpactGame {
+
+SpaceImpact::SpaceImpact(const ivec2& windowSize)
+: mWindowSize(windowSize),
+  mAsteroids(windowSize, NUM_ASTEROIDS),
+  mSpaceship(windowSize)
+ {
+ } 
+
+void SpaceImpact::Setup()
+{
+	mSpaceship.Setup([this](ShotHeight height) mutable
+	{
+		mAsteroids.TakeHit(height);
+		++mScore;
+	});
```

Comment:
The way you modeled Spaceship and asteroids is nice, but the shot with this mutable callback in between is a bit hacky. You don't need this callback. Why not just having another Entity (Shot, or Bullet or something). Then you can draw it separately (not in the Spaceship class! A bullet is not a Spaceship), and elegantly handle the collision between the bullet and the asteroid without having this kind of specific logic.